### PR TITLE
[IE CLDNN] 54304 fix reduce ops

### PR DESCRIPTION
**Pull request by andrei-cv**
### Details:
 - fix reduce ops nD->3D

### Tickets:
 - 54304

### CODE REVIEW

**Comment by vladimir-paramuzov** [comment]
@andrei-cv please add corresponding functional tests

File: inference-engine/src/cldnn_engine/ops/reduce.cpp [hunk 1]
```diff
@@ -89,7 +91,27 @@ void CreateReduceOp(Program& p, const std::shared_ptr<ngraph::Node>& op, cldnn::
         else if (rank - rawAxes.size() <= 4)
             out_format = cldnn::format::bfyx;
 
-        auto reorder_prim = cldnn::reorder(reorderLayerName, layerName, out_format, out_dt);
```
Comment (sshlyapn) [inline inference-engine/src/cldnn_engine/ops/reduce.cpp:92]: Is it correct to apply your fix under `if (!keep_dims && rank > 4)` condition? Could the same problem occur with `rank == 4`?

File: inference-engine/src/cldnn_engine/ops/reduce.cpp [hunk 2]
```diff
@@ -89,7 +91,27 @@ void CreateReduceOp(Program& p, const std::shared_ptr<ngraph::Node>& op, cldnn::
...
+        auto out_dims = op->get_output_shape(0).size();
+        if (out_dims == 3) {
+            resultLayerName = layerName + "_reshape";
+            auto out_shape = op->get_output_shape(0);
+            cldnn::tensor outTensor;
+            switch (rank) {
+                case 6:
+                    outTensor = cldnn::tensor(TensorValue(out_shape[0]), TensorValue(out_shape[1]),
+                                                1, TensorValue(out_shape[2]), 1, 1);
```
Comment (sshlyapn) [inline inference-engine/src/cldnn_engine/ops/reduce.cpp:102]: nit:
please align arguments as follow:
```
                    outTensor = cldnn::tensor(TensorValue(out_shape[0]), TensorValue(out_shape[1]),
                                              1, TensorValue(out_shape[2]), 1, 1);
```

**Comment by sshlyapn** [comment]
@andrei-cv please rebase and rerun CI

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**